### PR TITLE
Issue #19

### PR DESCRIPTION
**Pull request by s-cork**
files to look at:
`generated_parser.ts` 

`skulpt__parser_generator.py`

I also change `seq_flatten` to have a depth of `Infinity` - I think that's right?
If we try to do what the python generator does we get huge nested arrays. 
See the return values from 
https://github.com/python/cpython/blob/main/Tools/peg_generator/pegen/grammar_parser.py#L94
where most things are a tuple or a single element list

Here's the link to the code snippet in the py_generator
https://github.com/python/cpython/blob/44fb55149934d8fb095edb6fc3f8167208035b96/Tools/peg_generator/pegen/python_generator.py#L221-L235

And the c_generator
https://github.com/python/cpython/blob/main/Tools/peg_generator/pegen/c_generator.py#L691-L712

### CODE REVIEW

File: tools/gen_parser/skulpt_parser_genarator.py [hunk 1]
```diff
@@ -312,8 +314,12 @@ def visit_Alt(self, node: Alt, is_loop: bool, is_gather: bool) -> None:
                     if is_gather:
                         assert len(self.local_variable_names) == 2
                         action = f"[{self.local_variable_names[0]}, ...{self.local_variable_names[1]}]"
+                    elif len(self.local_variable_names) == 1:
+                        # @TODO is this correct?
+                        action = self.local_variable_names[0]
                     else:
-                        action = f"{', '.join(self.local_variable_names)}"
+                        # @TODO is it ok to return an array here
```
Comment (albertjan) [inline tools/gen_parser/skulpt_parser_genarator.py:321]: in the python version it's one if one variable and a tuple if more then one. This looks sensible to me. we'll see where the parser falls over 😄

File: src/parser/pegen.ts [hunk 2]
```diff
@@ -1391,7 +1392,7 @@ export function singleton_seq(p: Parser, a: AST): AST[] {
 
 /* Flattens an asdl_seq* of asdl_seq*s */
 export function seq_flatten(p: Parser, seqs: AST[][]): AST[] {
-    return seqs?.flat();
+    return seqs?.flat(Infinity) as AST[];
```
Comment (s-cork) [inline src/parser/pegen.ts:1395]: I'm not actually sure we need a dept of Infinity - Maybe we should leave as a depth of 1 and add this in if we find we need it?

Comment (s-cork) [reply src/parser/pegen.ts]: be7d1bd

**Comment by albertjan** [comment]
Looks good to me

**Pull request merged**